clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/operations/ChainResolver.java

```diff
@@ -116,11 +120,14 @@ public V value() {
         chainBuilder = chainBuilder.add(payload);
       }
     }
-    if(result != null) {
+    compacted = compacted || keyMatch > 1;
+    if(result == null) {
+      compacted = compacted || keyMatch > 0;
```

Comment:
I burned my brain for nearly an hour and I still don't understand why this line is 
`compacted = compacted || keyMatch > 0;`
and not just
`compacted = keyMatch > 0;`